### ISSUE TITLE
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()'

### ISSUE THREAD

**Issue by Boscop** [kbknapp/cargo-outdated#203]
Running it in my workspace doesn't seem to work:
```
cargo outdated -d 1
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', C:\Users\me\.cargo\registry\src\github.com-1ecc6299db9ec823\cargo-0.42.0\src/cargo\util\canonical_url.rs:42:13
stack backtrace:
   0: Ordinal0
   1: Ordinal0
   2: Ordinal0
   3: Ordinal0
   4: Ordinal0
   5: Ordinal0
   6: Ordinal0
   7: Ordinal0
   8: Ordinal0
   9: Ordinal0
  10: Ordinal0
  11: Ordinal0
  12: Ordinal0
  13: Ordinal0
  14: Ordinal0
  15: Ordinal0
  16: Ordinal0
  17: Ordinal0
  18: Ordinal0
  19: Ordinal0
  20: Ordinal0
  21: git_oidarray_free
  22: BaseThreadInitThunk
  23: RtlUserThreadStart
```

**Comment by deg4uss3r** [comment]
Hi @boscop, what version of `cargo-outdated` are you running (`cargo outdated —version`)? 

Thanks!

**Comment by Boscop** [comment]
@deg4uss3r 0.9.5

**Comment by deg4uss3r** [comment]
Thanks @Boscop, did you install as follows:

`cargo install --git https://github.com/kbknapp/cargo-outdated --force --locked`

I caught the `--locked` being left off the documentation this morning and updated that. We have to lock the `url` dependency that `cargo` is using to an older version until `ssh` parsing is resolved. Reinstall with above and let me know if you're still seeing this error.

**Comment by Boscop** [comment]
Thanks, that worked. So the url crate stopped accepting ssh urls?

When should I go back to using the crates.io version? When this https://github.com/kbknapp/cargo-outdated/issues/99 is closed?

**Comment by deg4uss3r** [comment]
No worries @Boscop 

It's a parsing issue, the `url` crate changed the way it does `ssh://` -> `https://` parsing and broke the `cargo` API for this. The next `cargo` update (`0.43.0`) will have a work around for the underlying `url` crate change. Once that happens we will test and patch into crates.io (which currently is out-of-date at the moment as is).

**Comment by Boscop** [comment]
@deg4uss3r Btw, do you have a link to the upstream issue? I can't seem to find it..

**Comment by deg4uss3r** [comment]
@Boscop I don't have an upstream issue; however, I did put in a PR for this https://github.com/rust-lang/cargo/pull/7849 and was told the issue was fixed in staging and awaiting the `v0.43.0` release. 

I hope that helps!
